### ISSUE TITLE
Sniffing adds invalid connections in 7.x

### ISSUE THREAD

**Issue by mo-getter** [elastic/elasticsearch-py#958]
When transport sniffing is enabled, invalid host names are added to the connection pool, which causes "Name or service not known" errors. This is caused by the reintroduction of `<hostname>/<ip>:<port>` format, which is the default as of 7.x+. The old syntax--which the sniffer parsing still expects--is now [deprecated](https://github.com/elastic/elasticsearch/blob/e1fdd00420b226b33adba3675ac0ff5d6daf954c/server/src/main/java/org/elasticsearch/http/HttpInfo.java#L89).

Related to #925.

**Comment by valentinomiazzo** [comment]
A workaround is to register a callback when you create the connection and do the fix there.
For example:

```
def _get_host_info(node_info, host):
        if node_info.get('roles', []) == ['master']:
            host = None
        if '/' in host.get('host'):
            host = {'host': host.get('host').rsplit('/', 1)[0], 'port': host.get('port')}
        return host

es = Elasticsearch(
                ...
                host_info_callback=_get_host_info,
            )
```

**Comment by LoadingZhang** [comment]
Thx @valentinomiazzo ,
`if '/' in host.get('host')` should change to `elif '/' in host.get('host')`, else get error on master node.

**Comment by HonzaKral** [comment]
Resolved by #1039
